wire-ios/WireUITests/PersonalUserTests.swift

```diff
@@ -0,0 +1,132 @@
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import XCTest
+
+final class PersonalUsersTests: XCTestCase {
+    var app: XCUIApplication!
+
+    override func setUpWithError() throws {
+        app = XCUIApplication()
+        app.launchArguments = [
+            "-BackendEnvironmentTypeOverrideKey staging",
+            "--preferred-api-version=8"
+        ]
+        app.useWireAuthentication()
+
+        app.launch()
+
+        // In UI tests it is usually best to stop immediately when a failure occurs.
+        continueAfterFailure = false
+    }
+
+    override func tearDownWithError() throws {
+        app = nil
+    }
+
+    @MainActor
+    func test_register_asPersonalUser() async throws {
+        let textField = emailTextField()
+        let email = "adfdasfasfaewa@wire.engineering"
+        textField.tap()
+        textField.typeText(email)
+
+        nextButton().tap()
+
+        createPersonalAccountLink().tap()
+
+        newNextButton().tap()
+
+        acceptButton().tap()
+
+        let verificationCode = try await getVerificationCode(email:email)
+
+        verificationCodeInput().tap()
+        verificationCodeInput().typeText(verificationCode)
+
+        let fullScreenshot = XCUIScreen.main.screenshot()
+        let screenshot = XCTAttachment(screenshot: fullScreenshot)
+        screenshot.lifetime = .keepAlways
+        add(screenshot)
+    }
+
+    // MARK: - Helpers
+
+    private func getVerificationCode(email:String) async throws -> String {
+        let inbucketURL = "https://\(ProcessInfo.processInfo.environment["INBUCKET_URL"]!)"
+        let inbucketUsername = ProcessInfo.processInfo.environment["INBUCKET_USERNAME"]!
+        let inbucketPassword = ProcessInfo.processInfo.environment["INBUCKET_PASSWORD"]!
+        var verificationCode:String = ""
+        let url = URL(string: "\(inbucketURL)/api/v1/mailbox/\(email)/latest")
+        guard let requestUrl = url else { fatalError() }
+        var request = URLRequest(url: requestUrl)
+        request.httpMethod = "GET"
+        let loginString = String(format: "%@:%@", inbucketUsername, inbucketPassword)
+        let loginData = loginString.data(using: String.Encoding.utf8)!
+        let base64LoginString = loginData.base64EncodedString()
+        request.setValue("Basic \(base64LoginString)", forHTTPHeaderField: "Authorization")
+
+        let (inbucketData,_) = try await URLSession.shared.data(for: request)
+
+        // Convert HTTP Response Data to a simple String
+        let message:InbucketMessage = try! JSONDecoder().decode(InbucketMessage.self, from:inbucketData)
+        let subject:String = message.subject
+        verificationCode = String(subject.prefix(6))
+    
+        print("Verification Code Found: \(verificationCode) for \(email)")
+        return verificationCode
```

Comment:
suggestion: move this to own helper class

Reply:
Yes, big agree and that is the plan. This draft PR was made mostly for showing the 1Password part to everyone - the rest of this is still such a work in progress